### PR TITLE
707 fill group to handle mc

### PR DESCRIPTION
**Pull request by AntonZogk**
# 707 fill group to handle mc

# Summary

This pull request fixes a bug in which references had wrong imputation flag (fimc instead of fir)
# Type of Change

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe):

# Checklists

This pull request meets the following requirements:

## Creator Checklist

- [x] Installable with all dependencies recorded
- [x] Runs without error
- [x] Follows PEP8 and project-specific conventions
- [x] Appropriate use of comments, for example, no descriptive comments
- [x] Functions documented using Numpy style docstrings
- [x] Assumptions and decisions log considered and updated if appropriate
- [x] Unit tests have been updated to cover essential functionality for a reasonable range of inputs and conditions
- [x] Other forms of testing such as end-to-end and user-interface testing have been considered and updated as required

If you feel some of these conditions do not apply for this pull request, please
add a comment to explain why.

## Reviewer Checklist

- [ ] Test suite passes (locally as a minimum)
- [ ] Peer reviewed with review recorded

# Additional Information

Please provide any additional information or context that would help the reviewer understand the changes in this pull request.

# Related Issues

Link any related issues or pull requests here.

### CODE REVIEW

**Comment by AntonZogk** [comment]
Here is the failed unit test https://github.com/ONSdigital/monthly-business-survey-results/actions/runs/12831190237/job/35781024197 for the uploaded scenario which should cover the fimc fir bug

**Comment by AntonZogk** [comment]
Removed a couple of rogue lines which were created for debugging also tidied up a bit the unit test

File: tests/imputation/test_imputation_flags.py [hunk 1]
```diff
@@ -82,6 +44,7 @@ def test_imputation_marker_manual_construction(
             reference="reference",
             strata="strata",
             auxiliary="auxiliary",
+            predictive_auxiliary="f_match_auxiliary",
```
Comment (lhubbardONS) [inline tests/imputation/test_imputation_flags.py:47]: Would be interested to know what purpose this has. I know it's not relevant to this review as it's for the other test cases, but I can't figure out where it it's being used in the function - I'm sure I'm missing something!

**Review by lhubbardONS — COMMENTED** [review]
Nice work - changes look good, and tests and hooks all pass. 

Just had one comment: looking at the test data, I wonder if there should be a case where an mc is followed by a return in the next period (as in the examples provided in the testing document)? Obviously please disregard if this case if already tested elsewhere. Would be good to get your thoughts :)

File: tests/imputation/test_imputation_flags.py (hunk 1 above)
Comment (AntonZogk) [reply tests/imputation/test_imputation_flags.py:47]: You aren't missing anything, good spot ! It's a rogue argument ! Will take it out.

**Comment by AntonZogk** [comment]
> Nice work - changes look good, and tests and hooks all pass.
> 
> Just had one comment: looking at the test data, I wonder if there should be a case where an mc is followed by a return in the next period (as in the examples provided in the testing document)? Obviously please disregard if this case if already tested elsewhere. Would be good to get your thoughts :)

Like your idea, it's important making the unit tests solid and strong and cover edge cases, i added some values in the unit test to cover that scenario with a new reference id to check the MC-R.

**Review by lhubbardONS — APPROVED** [review]
Changes look good - nice work and happy to approve!
